lib/ash_authentication/add_ons/confirmation.ex

```diff
@@ -0,0 +1,140 @@
+      extensions: [AshAuthentication]
+
+    attributes do
+      uuid_primary_key :id
+      attribute :email, :ci_string, allow_nil?: false
+    end
+
+    authentication do
+      api MyApp.Accounts
+
+      confirmation do
+        monitor_fields [:email]
+      end
+
+      strategies do
+        # ...
+      end
+    end
+  end
+  ```
+
+  ## Attributes
+
+  A `confirmed_at` attribute will be added to your resource if it's not already
+  present (see `confirmed_at_field` in the DSL documentation).
+
+  ## Actions
+
+  By default confirmation will add an action which updates the `confirmed_at`
+  attribute as well as retrieving previously stored changes and applying them to
+  the resource.
+
+  If you wish to perform the confirm action directly from your code you can do
+  so via the `AshAuthentication.Strategy` protocol.
+
+  ### Example
+
+      iex> strategy = Info.strategy!(Example.User, :confirm)
+      ...> {:ok, user} = Strategy.action(strategy, :confirm, %{"confirm" => confirmation_token()})
+      ...> user.confirmed_at >= one_second_ago()
+      true
+
+  ## Plugs
+
+  Confirmation provides a single endpoint for the `:confirm` phase.  If you wish
+  to interact with the plugs directly, you can do so via the
+  `AshAuthentication.Strategy` protocol.
+
+  ### Example
+
+      iex> strategy = Info.strategy!(Example.User, :confirm)
+      ...> conn = conn(:get, "/user/confirm", %{"confirm" => confirmation_token()})
+      ...> conn = Strategy.plug(strategy, :confirm, conn)
+      ...> {_conn, {:ok, user}} = Plug.Helpers.get_authentication_result(conn)
+      ...> user.confirmed_at >= one_second_ago()
+      true
+
+  ## DSL Documentation
+
+  #{Spark.Dsl.Extension.doc_entity(strategy(:confirmation))}
+  """
+
+  defstruct token_lifetime: nil,
+            monitor_fields: [],
+            confirmed_at_field: :confirmed_at,
+            confirm_on_create?: true,
+            confirm_on_update?: true,
+            inhibit_updates?: false,
+            sender: nil,
+            confirm_action_name: :confirm,
+            resource: nil,
+            provider: :confirmation,
+            name: :confirm
+
+  alias Ash.Changeset
+  alias AshAuthentication.{AddOn.Confirmation, Jwt}
+
+  @type t :: %Confirmation{
+          token_lifetime: hours :: pos_integer,
+          monitor_fields: [atom],
+          confirmed_at_field: atom,
+          confirm_on_create?: boolean,
+          confirm_on_update?: boolean,
+          inhibit_updates?: boolean,
+          sender: nil | {module, keyword},
+          confirm_action_name: atom,
+          resource: module,
+          provider: :confirmation,
+          name: :confirm
+        }
+
+  @doc """
+  Generate a confirmation token for a changeset.
+
+  This will generate a token with the `"act"` claim set to the confirmation
+  action for the strategy, and the `"chg"` claim will contain any changes.
+
+  FIXME: The "chg" claim should encrypt the contents of the changes so as to not
```

Comment:
we should make a GH issue for this, so this isn't lost.

Reply:
:+1: